code/game/machinery/hologram.dm

```diff
@@ -68,197 +104,350 @@ var/list/holopads = list()
+
+		if(LAZYLEN(holo_calls))
+			dat += "=====================================================<br>"
+
+		var/one_answered_call = FALSE
+		var/one_unanswered_call = FALSE
+		for(var/I in holo_calls)
+			var/datum/holocall/HC = I
+			if(HC.connected_holopad != src)
+				dat += "<a href='?src=[UID()];connectcall=\ref[HC]'>Answer call from [get_area(HC.calling_holopad)].</a><br>"
+				one_unanswered_call = TRUE
+			else
+				one_answered_call = TRUE
+
+		if(one_answered_call && one_unanswered_call)
+			dat += "=====================================================<br>"
+		//we loop twice for formatting
+		for(var/I in holo_calls)
+			var/datum/holocall/HC = I
+			if(HC.connected_holopad == src)
+				dat += "<a href='?src=[UID()];disconnectcall=\ref[HC]'>Disconnect call from [HC.user].</a><br>"
+
+
+	var/datum/browser/popup = new(user, "holopad", name, 300, 130)
+	popup.set_content(dat)
+	popup.set_title_image(user.browse_rsc_icon(icon, icon_state))
+	popup.open()
+
+/obj/machinery/hologram/holopad/Topic(href, href_list)
+	if(..() || isAI(usr))
+		return
+	add_fingerprint(usr)
+	if(stat & NOPOWER)
+		return
+	if(href_list["AIrequest"])
+		if(last_request + 200 < world.time)
 			last_request = world.time
-			to_chat(user, "<span class='notice'>You request an AI's presence.</span>")
+			temp = "You requested an AI's presence.<br>"
+			temp += "<a href='?src=[UID()];mainmenu=1'>Main Menu</a>"
 			var/area/area = get_area(src)
-			for(var/mob/living/silicon/ai/AI in living_mob_list)
-				if(!AI.client)	continue
-				to_chat(AI, "<span class='info'>Your presence is requested at <a href='?src=[AI.UID()];jumptoholopad=\ref[src]'>\the [area]</a>.</span>")
+			for(var/mob/living/silicon/ai/AI in ai_list)
+				if(!AI.client)
+					continue
+				to_chat(AI, "<span class='info'>Your presence is requested at <a href='?src=\ref[AI];jumptoholopad=[UID()]'>\the [area]</a>.</span>")
 		else
-			to_chat(user, "<span class='notice'>A request for AI presence was already sent recently.</span>")
-
+			temp = "A request for AI presence was already sent recently.<br>"
+			temp += "<a href='?src=[UID()];mainmenu=1'>Main Menu</a>"
+
+	else if(href_list["Holocall"])
+		if(outgoing_call)
+			return
+
+		temp = "You must stand on the holopad to make a call!<br>"
+		temp += "<a href='?src=[UID()];mainmenu=1'>Main Menu</a>"
+		if(usr.loc == loc)
+			var/list/callnames = list()
+			for(var/I in holopads)
+				var/area/A = get_area(I)
+				if(A)
+					LAZYADD(callnames[A], I)
+			callnames -= get_area(src)
+
+			var/result = input(usr, "Choose an area to call", "Holocall") as null|anything in callnames
+
+			if(qdeleted(usr) || !result || outgoing_call)
+				return
+
+			if(usr.loc == loc)
+				temp = "Dialing...<br>"
+				temp += "<a href='?src=[UID()];mainmenu=1'>Main Menu</a>"
+				new /datum/holocall(usr, src, callnames[result])
+
+	else if(href_list["connectcall"])
+		var/datum/holocall/call_to_connect = locateUID(href_list["connectcall"])
+		if(!qdeleted(call_to_connect))

```

Comment:
You should make sure that `src` is in `call_to_connect.dialing`. Make sure that `dialing` isn't `null`.

Also, obnoxiously, `qdeleted` returns `0` (`FALSE`) if its argument is `null` (fails the `istype` call), so you'll need to add a `null` check by hand, or modify `qdeleted` to return `1` if the `istype` check is failed.